catalog/transformer/catalog-transformer-xml/src/test/java/ddf/catalog/transform/xml/adapter/AttributeAdapterTest.java

```diff
@@ -0,0 +1,108 @@
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.AttributeType.AttributeFormat;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transformer.xml.adapter.AttributeAdapter;
+import ddf.catalog.transformer.xml.binding.AbstractAttributeType;
+import ddf.catalog.transformer.xml.binding.ObjectElement;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class AttributeAdapterTest {
+
+  @Test
+  public void testUnmarshalSeralizedObject() throws Exception {
+    AttributeAdapter attributeAdapter = getObjectAttributeAdapter();
+
+    Map<String, String> originalMap = ImmutableMap.of("key1", "val1", "key2", "val2");
+    Attribute attribute = new AttributeImpl("sampleAttr", (Serializable) originalMap);
+
+    AbstractAttributeType abstractAttributeType = attributeAdapter.marshal(attribute);
+    Attribute result = attributeAdapter.unmarshal(abstractAttributeType);
+
+    Map<String, String> resultMap = (Map) result.getValue();
+    assertThat(
+        "Maps are not equal.", resultMap.entrySet(), everyItem(isIn(originalMap.entrySet())));
+  }
+
+  @Test
+  public void testUnmarshalSeralizedObjectCollectionContainingNull() throws Exception {
+    AttributeAdapter attributeAdapter = getObjectAttributeAdapter();
+
+    List<String> originalList = Collections.unmodifiableList(Arrays.asList("val1", null, "val2"));
+
+    Attribute attribute = new AttributeImpl("sampleAttr", (Serializable) originalList);
+
+    AbstractAttributeType abstractAttributeType = attributeAdapter.marshal(attribute);
+    Attribute result = attributeAdapter.unmarshal(abstractAttributeType);
+
+    List<String> resultList = (List) result.getValues();
+    assertThat("Lists are not equal.", resultList, everyItem(isIn(originalList)));
+  }
+
+  @Test
+  public void testUnmarshalSeralizedObjectEmpty() throws Exception {
+    AttributeAdapter attributeAdapter = getObjectAttributeAdapter();
+
+    Attribute attribute = new AttributeImpl("sampleAttr", Collections.emptyList());
+
+    AbstractAttributeType abstractAttributeType = attributeAdapter.marshal(attribute);
+    Attribute result = attributeAdapter.unmarshal(abstractAttributeType);
+
+    assertThat("Null attribute not returned.", result == null);
+  }
+
+  @Test
+  public void testUnmarshalSeralizedObjectInvalidByteArray() throws Exception {
+    AttributeAdapter attributeAdapter = getObjectAttributeAdapter();
+
+    ObjectElement mockObjectElement = mock(ObjectElement.class);
+    doReturn("name").when(mockObjectElement).getName();
+    doReturn(ImmutableList.of(new byte[] {-31, 0, 0, 0})).when(mockObjectElement).getValue();
+    Attribute result = attributeAdapter.unmarshal(mockObjectElement);
+
+    assertThat("Unmarshalled byte array not returned", result.getValue() instanceof byte[]);
```

Comment:
✏️ 
```suggestion
    assertThat("Unmarshalled byte array not returned", result.getValue(), is(instanceOf(byte[])));
```

Reply:
This expects a class to be passed in, so I'm not sure if I'm able to use it in this case

Reply:
`byte[].class`

Reply:
Could've sworn that was the first thing I tried but I guess I didn't 😄